ext/mixed/js/display.js

```diff
@@ -357,18 +355,14 @@ class Display {
 
     async setContentTerms(definitions, context, token) {
         if (!context) { throw new Error('Context expected'); }
-        if (!this.isInitialized()) { return; }
 
         this.setEventListenersActive(false);
 
         if (context.focus !== false) {
             window.focus();
         }
 
-        if (!this.displayGenerator.isInitialized()) {
-            await this.displayGenerator.initialize();
-            if (this.setContentToken !== token) { return; }
-        }
+        if (this.setContentToken !== token) { return; }
```

Comment:
Just noticed that this is probably not needed unless there's something async before coming into the method. I'll look at it later. The `promiseTimeout` below still needs the token check.

Reply:
Fixed